### DISCUSSION TITLE
Questions on returning vecs and including luarocks modules in mlua 0.10

### DISCUSSION THREAD

**Discussion by cheesycod** [mlua-rs/mlua#432] [General]
Just a quick question, how would I deserialize a Lua array in mlua since serde complains about ``invalid type: map, expected a sequence``. Also, is it possible to include modules from luarocks or similar in mlua (luau) somehow?

**Comment by khvzak** [comment]
> Just a quick question, how would I deserialize a Lua array in mlua since serde complains about `invalid type: map, expected a sequence`.

Do you have a code example?

>  Also, is it possible to include modules from luarocks or similar in mlua (luau) somehow?

You can build modules for luau in mlua and use them. I don't think luarocks support luau, as it's a completely independent ecosystem

**Reply by cheesycod** [reply]
> Do you have a code example?

```rust
#[derive(Serialize, Deserialize, Debug, Default, Clone)]
pub struct MessageEmbed {
    /// The title set by the template
    pub title: Option<String>,
    /// The description set by the template
    pub description: Option<String>,
    /// The fields that were set by the template
    pub fields: Vec<MessageEmbedField>,
}
```

In Lua:

local embed = { title = "", description = "", fields = {} }

Leads to the above error. Or do I have to set the array metaclass on fields for this to work?

> 
> > Also, is it possible to include modules from luarocks or similar in mlua (luau) somehow?
> 
> You can build modules for luau in mlua and use them. I don't think luarocks support luau, as it's a completely independent ecosystem

Ah alr

**Reply by khvzak** [reply]
>  do I have to set the array metaclass on fields for this to work?

Yes. `fields = setmetatable({}, array_mt)` where `array_mt` is https://docs.rs/mlua/0.10.0-beta.1/mlua/struct.Lua.html#method.array_metatable

**Reply by khvzak** [reply]
See https://github.com/openresty/lua-cjson?tab=readme-ov-file#array_mt where the idea came from
